libs/ts-rest/core/src/lib/client.ts

```diff
@@ -114,7 +114,7 @@ export const defaultApi: ApiFetcher = async ({
   try {
     return {
       status: result.status,
-      body: await result.json(),
+      body: await result.clone().json(),
```

Comment:
any memory implications with this? 🤔

Reply:
Yeah, based on [this](https://developer.mozilla.org/en-US/docs/Web/API/Response/clone)

> If only one cloned branch is consumed, then the entire body will be buffered in memory.

But, I'm pretty sure the GC will take care of it.

Reply:
Or to just get rid of any doubt maybe we can read it once as text, then `try { return JSON.parse(text); } catch { return text }`

Reply:
Yeah, not sure if we should rely on the already expensive GC that is Javascript/Node LOL.